typescript/cli/src/deploy/warp.ts

```diff
@@ -346,13 +346,92 @@ async function createWarpIsm(
       staticMerkleRootWeightedMultisigIsmFactory,
       staticMessageIdWeightedMultisigIsmFactory,
     },
-    config: warpConfig[chain].interchainSecurityModule!,
+    config: interchainSecurityModule,
     contractVerifier,
   });
   const { deployedIsm } = evmIsmModule.serialize();
   return deployedIsm;
 }
 
+async function createWarpHook({
+  chain,
+  context,
+  contractVerifier,
+  warpConfig,
+  ismFactoryDeployer,
+}: {
+  chain: string;
+  context: WriteCommandContext;
+  contractVerifier?: ContractVerifier;
+  warpConfig: TokenRouterConfig;
+  ismFactoryDeployer: HyperlaneProxyFactoryDeployer;
+}): Promise<HookConfig | undefined> {
+  const { hook } = warpConfig;
+
+  if (!hook || typeof hook === 'string') {
+    logGray(`Config Hook is ${!hook ? 'empty' : hook}, skipping deployment.`);
+    return hook;
+  }
+
+  logBlue(`Loading registry factory addresses for ${chain}...`);
+  let chainAddresses = await context.registry.getChainAddresses(chain);
+
+  if (!chainAddresses) {
+    logGray(`Registry factory addresses not found for ${chain}. Deploying...`);
+    chainAddresses = serializeContracts(
+      await ismFactoryDeployer.deployContracts(chain),
+    ) as Record<string, string>;
+  }
+
+  logGray(`Creating ${hook.type} Hook for token on ${chain} chain...`);
+
+  const {
+    mailbox,
+    domainRoutingIsmFactory,
+    staticAggregationHookFactory,
+    staticAggregationIsmFactory,
+    staticMerkleRootMultisigIsmFactory,
+    staticMessageIdMultisigIsmFactory,
+    staticMerkleRootWeightedMultisigIsmFactory,
+    staticMessageIdWeightedMultisigIsmFactory,
+  } = chainAddresses;
+  const proxyFactoryFactories = {
+    domainRoutingIsmFactory,
+    staticAggregationHookFactory,
+    staticAggregationIsmFactory,
+    staticMerkleRootMultisigIsmFactory,
+    staticMessageIdMultisigIsmFactory,
+    staticMerkleRootWeightedMultisigIsmFactory,
+    staticMessageIdWeightedMultisigIsmFactory,
+  };
+
+  // If config.proxyadmin.address exists, then use that. otherwise deploy a new proxyAdmin
+  const proxyAdminAddress: Address =
+    warpConfig.proxyAdmin?.address ??
+    (
+      await context.multiProvider.handleDeploy(
+        chain,
+        new ProxyAdmin__factory(),
+        [],
+      )
+    ).address;
+
+  const evmHookModule = await EvmHookModule.create({
+    chain,
+    multiProvider: context.multiProvider,
+    coreAddresses: {
+      mailbox,
+      proxyAdmin: proxyAdminAddress,
+    },
+    config: hook,
+    contractVerifier,
+    proxyFactoryFactories,
+  });
```

Comment:
this should work too
```suggestion
  // If config.proxyadmin.address exists, then use that. otherwise deploy a new proxyAdmin
  const proxyAdminAddress: Address =
    warpConfig.proxyAdmin?.address ??
    (
      await context.multiProvider.handleDeploy(
        chain,
        new ProxyAdmin__factory(),
        [],
      )
    ).address;

  const evmHookModule = await EvmHookModule.create({
    chain,
    multiProvider: context.multiProvider,
    coreAddresses: {
      mailbox,
      proxyAdmin: proxyAdminAddress,
    },
    config: hook,
    contractVerifier,
    chainAddresses,
  });
```

Reply:
doesn't work :/